### PR TITLE
Making the ThinkDifferent classes be JDK version-independent. JDK 9…

### PR DESCRIPTION
**Pull request by ssilverman**
… and later have a new java.awt.desktop package that replaces the custom handlers in the com.apple.eawt package. This commit uses java.lang.reflect.Proxy to generate handlers of the correct type, independent of the handler parameter types, which are different across JDK versions.

Specifically, this fixes being able to run a Processing app (externally or otherwise) on OSX using a JDK having a version > 1.8 (i.e. 9 and above).

### CODE REVIEW

**Comment by ssilverman** [comment]
Fixes bug: #5746.

File: app/src/processing/app/platform/ThinkDifferent.java [hunk 1]
```diff
@@ -131,12 +131,51 @@ public void handleQuitRequestWith(QuitEvent event, QuitResponse response) {
...
   }
-  
+
 
 //  public ThinkDifferent(Base base) {
 //    this.base = base;
 //  }
 
+  /**
+   * Sets a handler on an instance of {@link Application}, taking into account JVM version
```
Comment (ssilverman) [inline app/src/processing/app/platform/ThinkDifferent.java]: Should say "JDK version", but I don't want to make a new commit.

Comment (benfry) [reply app/src/processing/app/platform/ThinkDifferent.java]: No, JVM is correct.

Comment (ssilverman) [reply app/src/processing/app/platform/ThinkDifferent.java]: My thinking on this is that the JDK includes the JVM plus libraries and other misc. stuff. You could theoretically change package names in the JDK 1.8 class files to JDK11-compliant names (eg. change `com.apple.eawt` to `java.awt.desktop`), and then run them on an older JVM (subject to only non-native code, etc). So my view is that this is still a JDK difference and not a JVM difference. JVM, in my mind, includes bytecodes plus support, and doesn't include the libraries (with some exceptions, eg. invokedynamic implementation).

Comment (ssilverman) [reply app/src/processing/app/platform/ThinkDifferent.java]: To play devil's advocate with myself: Originally, "Java" was defined as a JVM plus libraries plus language, so in reality, I think the lines are kind of fuzzy anyway. I won't object to whatever's chosen, "JVM" or "JDK".

**Comment by benfry** [comment]
Fixed with https://github.com/processing/processing4/pull/1 and continued development happening at https://github.com/processing/processing4

**Pull request closed without merging**